source/agent/addons/quic/QuicTransportConnection.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef QUIC_QUICTRANSPORTCONNECTION_H_
+#define QUIC_QUICTRANSPORTCONNECTION_H_
+
+#include <memory>
+#include <mutex>
+#include <unordered_map>
+#include <vector>
+
+#include <logger.h>
+#include <nan.h>
+
+#include "QuicTransportStream.h"
+#include "owt/quic/quic_transport_session_interface.h"
+
+class QuicTransportConnection : public Nan::ObjectWrap, public owt::quic::QuicTransportSessionInterface::Visitor, QuicTransportStream::Visitor {
+    DECLARE_LOGGER();
+
+public:
+    class Visitor {
+        // Authentication passed with user ID `id`.
+        virtual void onAuthentication(const std::string& id) = 0;
+        // Connection is closed.
+        virtual void onClose() = 0;
+    };
+    explicit QuicTransportConnection();
+    ~QuicTransportConnection();
+    void setVisitor(Visitor* visitor);
+    static v8::Local<v8::Object> newInstance(owt::quic::QuicTransportSessionInterface* session);
+
+    static NAN_MODULE_INIT(init);
+    static NAN_METHOD(newInstance);
+    static NAN_METHOD(createBidirectionalStream);
+    static NAUV_WORK_CB(onStreamCallback);
+
+    static Nan::Persistent<v8::Function> s_constructor;
+
+protected:
+    // Overrides owt::quic::QuicTransportSessionInterface::Visitor.
+    void OnIncomingStream(owt::quic::QuicTransportStreamInterface*) override;
+    void OnCanCreateNewOutgoingStream(bool unidirectional) override { }
+
+    // Overrides QuicTransportStream::Visitor.
+    void onEnded() override;
+
+private:
+    owt::quic::QuicTransportSessionInterface* m_session;
```

Comment:
Use std::unique_ptr<QuicTransportSessionInterface>

Reply:
It's raw pointer because `QuicTransportConnection` doesn't own `m_session`.